table_schema_cache.go

```diff
@@ -211,6 +209,70 @@ func LoadTables(db *sql.DB, tableFilter TableFilter, columnCompressionConfig Col
 	return tableSchemaCache, nil
 }
 
+func (ts *TableSchema) findColumnByName(name string) *schema.TableColumn {
+	for _, column := range ts.Columns {
+		if column.Name == name {
+			return &column
+		}
+	}
+	return nil
+}
+
+func LackPaginationKeyError(schema, table string, hasCompositePrimaryKey bool) error {
+	if hasCompositePrimaryKey {
+		return fmt.Errorf("no Pagination Key Column provided for %s.%s which has a composite primary key. Attempt to fallback to `id` column failed as the same is not existent. Kindly specify a Pagination Key Column", schema, table)
+	}
+	return fmt.Errorf("no Pagination Key Column provided for %s.%s which has no Primary Key to fallback to. Attempt to fallback to `id` column failed as the same is not existent. Kindly specify a Pagination Key Column", schema, table)
+}
+
+func PaginationKeyNotFoundError(schema, table, paginationColumn string) error {
+	return fmt.Errorf("Pagination Key Column provided: %s for %s.%s is a non existing column", paginationColumn, schema, table)
+}
+
+func (ts *TableSchema) setPaginationKeyColumn(dbname, table string, paginateByColumnConfig PaginateByColumnConfig) error {
+	var err error
+	if paginationColumn, found := paginateByColumnConfig.PaginationColumnFor(dbname, table); found {
+		ts.PaginationKeyColumn = ts.findColumnByName(paginationColumn)
+		err = PaginationKeyNotFoundError(dbname, table, paginationColumn)
+	} else {
+		switch len(ts.PKColumns) {
+		case 0:
+			ts.PaginationKeyColumn = ts.findColumnByName("id")
+			err = LackPaginationKeyError(dbname, table, false)
+		case 1:
+			ts.PaginationKeyColumn = &ts.Columns[ts.PKColumns[0]]
+		default:
+			ts.PaginationKeyColumn = ts.findColumnByName("id")
+			err = LackPaginationKeyError(dbname, table, true)
+		}
+	}
+	if ts.PaginationKeyColumn == nil {
+		return err
+	}
+
+	return nil
+}
+
+/*
+GetPKColumn is a method originally belonging to go-mysql/schema.Table
+and stands for: `Get Primary Key Column`
+
+As `TableSchema` embeds `schema.Table`, previously the method call to
+`TableSchema` would delegate automatically to the method for
+`schema.Table`. This was in spirit with the initial assumption that the
+column to iterate over in any given table is the Primary Key Column.
+
+In order to accommodate user provided pagination columns through
+configuration in the least painful way, this method has been
+reimplemented. Now this method stands for: `Get Pagination Key Column`.
+
+The param `index` currently serves no purpose but to maintain the same
+method signature as the original method.
+*/
+func (ts *TableSchema) GetPKColumn(index int) *schema.TableColumn {
+	return ts.PaginationKeyColumn
+}
```

Comment:
This is clever but easily confused with primary key. We discussed this in Slack. You suggested two followup cleanup PRs but would prefer to not work on the follow-ups yourself:

* Rename this method to something like `GetPaginationColumn`
* Rename references like `minPK`, `maxPK`, etc

Reply:
Will address this in the next PR

Reply:
I agree with @hkdsun on this point.